src/plugins/explore/public/components/tabs/action_bar/patterns_settings/patterns_settings_popover_content.tsx

```diff
@@ -7,7 +7,8 @@ import { EuiCallOut, EuiLoadingSpinner, EuiSelect, EuiText } from '@elastic/eui'
 import React from 'react';
 import { FormattedMessage } from '@osd/i18n/react';
 import { useDispatch, useSelector } from 'react-redux';
-import { DataView } from 'src/plugins/data/common';
+import { DataView } from 'src/plugins/data/common/data_views';
+import { IndexPatternField } from 'src/plugins/data/common';
```

Comment:
you should be able to use dataviewfield and update the import on line 10

Reply:
What would I need dataviewfield for? The only reason I'm using DataView here is to specify a parameter type

Reply:
so `DataViewField` extends `IndexPatternField` like how `DataView` extends `IndexPattern`.

Explore should use `DataView`. You are able to use the IndexPatternField interchangeably because it extends but for simplicity if you are already importing DataView, then you should be able to just do  
```
import { DataView. DataViewField } from 'src/plugins/data/common/data_views';
```

Reply:
It's not really working because I'm getting fields through dataset.fields.getAll(), which gives me a field list of type `IndexPatternField[]`. Is there some other way of getting the fields which will be of type `DataViewField`? Even though these types seem to pretty much be identical, they dont seem to have any relation for me to try and cast it over.